backend/layers/processing/dataset_metadata_update.py

```diff
@@ -0,0 +1,305 @@
+            cxg_metadata_dict.update(metadata_update.as_dict_without_none_values())
+
+        with tiledb.open(array_name, mode="w", ctx=ctx) as metadata_array:
+            metadata_array.meta["corpora"] = json.dumps(cxg_metadata_dict)
+
+        self.business_logic.add_dataset_artifact(dataset_version_id, DatasetArtifactType.CXG, new_cxg_dir)
+        self.update_processing_status(dataset_version_id, DatasetStatusKey.CXG, DatasetConversionStatus.CONVERTED)
+
+
+class DatasetMetadataUpdater(ProcessDownload):
+    def __init__(
+        self, business_logic: BusinessLogic, artifact_bucket: str, cellxgene_bucket: str, datasets_bucket: str
+    ) -> None:
+        super().__init__(business_logic, business_logic.uri_provider, business_logic.s3_provider)
+        self.artifact_bucket = artifact_bucket
+        self.cellxgene_bucket = cellxgene_bucket
+        self.datasets_bucket = datasets_bucket
+
+    def update_metadata(
+        self,
+        current_dataset_version_id: DatasetVersionId,
+        new_dataset_version_id: DatasetVersionId,
+        metadata_update: DatasetArtifactMetadataUpdate,
+    ):
+        current_dataset_version = self.business_logic.get_dataset_version(current_dataset_version_id)
+        if current_dataset_version.status.processing_status != DatasetProcessingStatus.SUCCESS:
+            self.logger.info(
+                f"Dataset {current_dataset_version_id} is not successfully processed. Skipping metadata update."
+            )
+            return
+
+        artifact_uris = {artifact.type: artifact.uri for artifact in current_dataset_version.artifacts}
+
+        new_dataset_version = self.business_logic.get_dataset_version(new_dataset_version_id)
+        if new_dataset_version.status.processing_status != DatasetProcessingStatus.INITIALIZED:
+            self.logger.info(
+                f"Dataset {new_dataset_version_id} has processing status "
+                f"{new_dataset_version.status.processing_status} rather than expected INITIALIZED."
+                f"Skipping metadata update."
+            )
+            return
+
+        if DatasetArtifactType.RAW_H5AD in artifact_uris:
+            raw_h5ad_uri = artifact_uris[DatasetArtifactType.RAW_H5AD]
+        else:
+            self.logger.error(f"Cannot find raw H5AD artifact uri for {current_dataset_version_id}.")
+            raise ValueError
+
+        # uploads current dataset version's raw h5ad to artifact bucket directory for the new dataset version
+        self.process(new_dataset_version_id, raw_h5ad_uri, self.artifact_bucket)
+
+        new_artifact_key_prefix = self.get_key_prefix(new_dataset_version_id.id)
+
+        artifact_jobs = []
+
+        with multiprocessing.Pool(processes=3) as pool:
+            if DatasetArtifactType.H5AD in artifact_uris:
+                self.logger.info("Main: Starting thread for h5ad update")
+                artifact_jobs.append(
+                    pool.apply_async(
```

Comment:
Get rid of the pool and just use `Process` and `.start()` it.

Replace `.get()` below with `.join()`.